### ISSUE TITLE
fsample in ft_datatype_raw

### ISSUE THREAD

**Issue by NirOfir** [fieldtrip/fieldtrip#1837]
I'm trying to convert an FT dataset to eeglab (so I can then use LIMO on it), and I saw that fieldtrip2eeglab looks for an "Fs" field in the data structure. I got to datatype_raw to see what this field is supposed to be called, and I saw that the code sets by default an "fsample" field. However, in the help section, it is written as "deprecated". What does deprecated mean here? From the code, it seems more like a required field to me.

Thanks!

**Comment by schoffelen** [comment]
In FT raw datastructures the sampling rate is called (if present) 'fsample', in the header (after ft_read_header), it's called 'Fs', I think. fieldtrip2eeglab is probably not really up-to-date, so any glitches there are probably the consequence of poor code quality, rather than you doing something wrong.

From the point-of-view of fieldtrip, fsample should not be needed, because it can in principle be derived from the time-axis of the trials, i.e. 1./(mean(diff(data.time{1}))). Indeed I am a bit confused myself as to why it is mentioned to be deprecated, but still computed (and added to the data structure). I think this is the consequence of the fact that the intended deprecatedness of fsample has never made it into the complete codebase (if I grep for fsample in the main fieldtrip functions, it is quite often still assumed to be present as a field in the data).

**Comment by NirOfir** [comment]
Yes, fsample and Fs are what I found too. I'm trying to move forward with fieldtrip2eeglab through submitting issues over there.

So, there are 2 main possibilities I think (if you don't want to just keep everything as it is):
1. Make fsample really deprecated - remove it from ft_datatype_raw, and have all other functions that use raw types do this calculation themselves.
2. Keep ft_datatype_raw as it is, and make sure all functions rely on the fsample field.

I suppose solution 1 is preferred, correct?

**Comment by schoffelen** [comment]
I would for the moment really focus on getting fieldtrip2eeglab more robust, e.g. by not relying on a hdr.Fs, or a data.fsample, but allow for the possibility to estimate the fsample from the data, if it cannot be harvested from hdr or data directly.

**Comment by NirOfir** [comment]
Perfect, so I'm focusing on sccn/eeglab#356.

Thanks again!

**Comment by schoffelen** [comment]
https://github.com/sccn/eeglab/pull/396

This should address the current issue
